app/controllers/bot_instances_controller.rb

```diff
@@ -2,18 +2,21 @@ class BotInstancesController < ApplicationController
   before_action :authenticate_user!, :only => [:new, :create, :edit, :update, :destroy]
   before_action :set_bot_instance, only: [:show, :edit, :update, :destroy]
   before_action :check_instance_permissions, only: [:edit, :update, :destroy]
-  before_action :check_bot_permissions, only: [:new, :create]
```

Comment:
You could create a `set_bot` callback, on the appropriate actions, and put it before `check_bot_permissions` in the filter chain to solve this.